tasks/Task.cpp

```diff
@@ -107,14 +111,13 @@ void Task::updateHook()
     TaskBase::updateHook();
 
     //read input ports
-    std::vector<base::Vector2d> corners;
+    std::vector<apriltags::VisualFeaturePoint> corners_vector;
     base::LinearAngular6DCommand setpoint;
-    bool no_corners = (_marker_corners.read(corners) == RTT::NoData);
-    bool no_setpoint = (_cmd_in.read(setpoint) == RTT::NoData);
-    ctrl_state.timestamp = base::Time::now();
+    bool no_corners = (_marker_corners.read(corners_vector) != RTT::NewData);
+    bool no_setpoint = (_cmd_in.read(setpoint) != RTT::NewData);
```

Comment:
Why do you require a new setpoint each time ?

Apart from breaking general conventions (unless in a nested control loop, setpoints are either sent only once, or at low frequency), it creates a race condition between the corners and setpoint: you could read the setpoint in one updateHook and the corner in another, leading to not doing anything.